File: covid-api/src/main/java/com/tacs/grupo1/covid/api/dto/statistics/CountryPlot.java
package com.tacs.grupo1.covid.api.dto.statistics;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.tacs.grupo1.covid.api.dto.Country;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
public class CountryPlot {
    private Country country;
    @JsonProperty("statistics_day_list")
    private List<StatisticsDay> statisticsDayList;
}

File: covid-frontend/src/models/CountryPlot.ts
import Country from "./Country";
import StatisticsDay from "./StatisticsDay";

export default class CountryPlot{
    public country: Country;
    public statistics_day_list: Array<StatisticsDay>;

    constructor(countryPlot?: any){
        if(countryPlot){
            this.country = countryPlot.country;
            this.statistics_day_list = countryPlot.statistics_day_list;

        }else{
            this.country = new Country();
            this.statistics_day_list = new Array<StatisticsDay>();
        }
    }
}